riscv/csrs.cc

```diff
@@ -638,11 +638,7 @@ void sstatus_csr_t::dirty(const reg_t dirties) {
 }
 
 bool sstatus_csr_t::enabled(const reg_t which) {
-  if ((orig_sstatus->read() & which) != 0) {
-    if (!state->v || (virt_sstatus->read() & which) != 0)
-      return true;
-  }
-  return false;
+  return ((orig_sstatus->read() & which) != 0) && (!state->v || (virt_sstatus->read() & which) != 0);
```

Comment:
IMO the original version of this function is easier to read (even though it is verbose).

Reply:
Remove the second commit.